src/akvo_reflow/import.clj

```diff
@@ -0,0 +1,79 @@
+(ns akvo-reflow.import
+  (:require [akvo.commons.gae :as gae]
+            [akvo.commons.gae.query :as q]
+            [hugsql.core :as hugsql])
+  (:import [org.postgresql.util PGobject PSQLException]
+           com.fasterxml.jackson.databind.ObjectMapper))
+
+(def object-mapper (ObjectMapper.))
+
+(def batch-size 300)
+
+(def kinds (array-map "SurveyGroup" "survey"
+                      "Survey" "form"
+                      "QuestionGroup" "question_group"
+                      "Question" "question"
+                      "DeviceFiles" "device_file"
+                      "SurveyedLocale" "data_point"
+                      "SurveyInstance" "form_instance"
+                      "QuestionAnswerStore" "answer"))
+
+(hugsql/def-db-fns "akvo_reflow/import.sql" {:quoting :ansi})
+
+(defn entity->jsonb
+  [entity]
+  (doto (PGobject.)
+    (.setType "jsonb")
+    (.setValue (.writeValueAsString object-mapper entity))))
+
+(defn insert-entities
+  [db-spec table-name entities]
+  (doseq [e entities]
+    (let [created-datetime (.getProperty e "createdDateTime")
+          created-at (if (nil? created-datetime) 0 (.getTime created-datetime))]
+      (try
+        (insert-entity db-spec {:table-name table-name
+                                :created-at created-at
+                                :payload (entity->jsonb e)})
+        (catch PSQLException e
+          (.printStackTrace e))))))
```

Comment:
Yes, the intention is to avoid inserting duplicates. We get batches of 300 entities, I would like to be _granular_ and insert per entity instead of a transactions with the whole batch